### PR TITLE
Refactor into submodules

### PR DESCRIPTION
**Pull request by meshy**
This is mostly just moving code around, but does add a "SUCCESS" return-code, and do a couple of renames.

I intend to add more commands (as a re-work of https://github.com/Memrise/mypy-json-report/pull/18), and getting this out of the way will make that simpler.

### CODE REVIEW

File: mypy_json_report/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2022 Memrise
```
Comment (meshy) [inline mypy_json_report/__init__.py:1]: I wasn't really sure what to do with the copyright years here. Should they be 2024? 2022-2024? As-is?

File: mypy_json_report/return_codes.py
Comment (Tenzer) [inline mypy_json_report/return_codes.py:1]: I think "exit status" or "exit code" usually would be the term used for the code a program uses to signal its status when it returns: https://en.wikipedia.org/wiki/Exit_status. Could you use one of those terms instead please?

File: mypy_json_report/return_codes.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import enum
+
+
+class ReturnCode(enum.IntEnum):
+    SUCCESS = 0
+    # 1 is returned when an uncaught exception is raised.
+    # Argparse returns 2 when bad args are passed.
```
Comment (Tenzer) [inline mypy_json_report/return_codes.py:21]: Perhaps these should be added as regular entries, but we can keep the comments to expand further:
```suggestion
    # 1 is returned when an uncaught exception is raised.
    UNCAUGHT_EXCEPTION = 1
    # Argparse returns 2 when bad args are passed.
    ARGUMENT_PARSING_ERROR = 2
```

File: mypy_json_report/__init__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .cli import main
+
+
+if __name__ == "__main__":
+    main()
```
Comment (Tenzer) [inline mypy_json_report/__init__.py:19]: **Suggestion:** This should probably be moved to `__main__.py`? That's the file which is executed if you are to run `python -m mypy_json_report`. I don't think it really serves any purpose here.

**Review by Tenzer — COMMENTED** [review]
Looks good.

You should change the entrypoint for the package so it points at the new location for the `main` function: https://github.com/Memrise/mypy-json-report/blob/94a9aa0e03ad74a3b28c6d58a33dc75eb4530819/pyproject.toml#L37

File: mypy_json_report/__init__.py (hunk 1 above)
Comment (Tenzer) [reply mypy_json_report/__init__.py:1]: It should be the year the project was established, so keep it at 2022.

**Comment by meshy** [comment]
Thanks @Tenzer. I think I've addressed your points now.

**Review by Tenzer — APPROVED** [review]
Thanks!

**Pull request merged**